applications/Chat/coati/dataset/sft_dataset.py

```diff
@@ -71,6 +71,29 @@ def _preprocess(sources: Sequence[str],
     return sequences_token["input_ids"], labels, sequences_token["attention_mask"]
 
 
+def _preprocess_chatglm(sources: Sequence[str],
+                targets: Sequence[str],
+                tokenizer: PreTrainedTokenizer,
+                max_length: int,
+                ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    """Preprocess the data by tokenizing."""
+    sequences = [s + "[gMASK]" + tokenizer.bos_token + t for s, t in zip(sources, targets)]
+    sequences_token = tokenizer(sequences,
```

Comment:
Why is it necessary to add [gMask] here, do we have any reference?

Reply:
gMASK is the special token in ChatGLM